include/snowplow/http/http_client_curl.hpp

```diff
@@ -13,7 +13,7 @@ See the Apache License Version 2.0 for the specific language governing permissio
 
 #ifndef HTTP_CLIENT_CURL_H
 #define HTTP_CLIENT_CURL_H
-#if !defined(WIN32) && !defined(_WIN32) && !defined(__WIN32) || defined(__CYGWIN__)
+#if !defined(__APPLE__) && !defined(WIN32) && !defined(_WIN32) && !defined(__WIN32) || defined(__CYGWIN__)
```

Comment:
I agree that we don't need to provide the CURL client on macOS as it just complicates the build, but it is a breaking change from the previous version so we'll need to change the tracker version to v2. I was wondering if you know of any other breaking changes in this PR that we should mention in the upgrading guide to the next version?

Reply:
Let me know if we should put back CURL so you need to raise a major version. I'll think this over if there are any other changes.

Reply:
Other possible breaking changes:

- CMake minimum version changed 3.14 -> 3.15
- As you mentioned the libcurl has been removed as a dependency of Snowplow on Macos. If a client project (using Snowplow as a subdirectory) relies on getting libcurl through Snowplow then it will fail now.
- A bit similar: sqlite3, libcurl, libuuid are now PRIVATE dependencies. If a client project relies on getting the sqlite3/libcurl/libuuid include paths through Snowplow then it will fail now.

Reply:
Thanks for listing the other breaking changes! I think it's fine to raise the major version in this case.